rfc/0016-instances-vs-types/0016-instances-vs-types.md

```diff
@@ -0,0 +1,123 @@
+		columns: [
+			"mpg" oftype decimal,
+			"cyl" oftype integer,
+			"disp" oftype decimal,
+		];
+	}
+	block CarsLoader oftype SQLiteLoader {
+		table: "Cars";
+		file: "./cars.sqlite";
+	}
+}
+```
+
+All block definitions (`block ... oftype ...`) are **definitions** of e.g., datasources or transformations (e.g. the `block CarsCSVExtractor oftype CSVExtractor` with all defined properties is a definition of the CSV file as a datasource).
+
+Instantiation happens only implicitly when a block reference is used in a pipeline, e.g. `CarsCSVExtractor -> CarsTableInterpreter` **instantiates** one instance of the `CarsCSVExtractor` during runtime. One block **definition** can be **instantiated** this way mutliple times in one pipeline.
+
+### Proposed change
+
+Allow **definitions** only outside of pipelines, allow block **instantiations** only inside of pipelines.
+
+Because pipelines get executed implicitly when executing a Jayvee model (and therefore pipelines get **instantiated** during runtime), it makes sense to bundle all other instantiations in them as well. This means everything outside of a pipeline is a **definition** (**type**), everything inside a pipeline is an **instance**.
+
+New syntax (minimal cars example):
+```
+// Pipeline containing block instantiations
+pipeline CarsPipeline {
+	CarsCSVExtractor 
+    -> CarsTableInterpreter
+		-> CarsLoader;
+}
+
+// Block definitions outside of pipeline
+block CarsCSVExtractor oftype CSVExtractor {
+  url: "https://gist.githubusercontent.com/noamross/e5d3e859aa0c794be10b/raw/b999fb4425b54c63cab088c0ce2c0d6ce961a563/cars.csv";
+  enclosing: '"';
+}
+block CarsTableInterpreter oftype TableInterpreter {
+  header: true;
+  columns: [
+    "mpg" oftype decimal,
+    "cyl" oftype integer,
+    "disp" oftype decimal,
+  ];
+}
+block CarsLoader oftype SQLiteLoader {
+  table: "Cars";
+  file: "./cars.sqlite";
+}
+```
+
+### Additional upsides
+Defining blocks only outside a pipeline also means we can reuse them across many pipelines. This has a positive side effect on blocks of types without non-default properties, such as the `TextFileInterpreter`. Before, any pipeline needed to define these blocks before it could instantiate them, leading to very verbose code.
+
+With this change, one definition of a generic `TextFileInterpreter` (e.g., `block StandardTextFileInterpreter oftype TextFileInterpreter { }`) outside of a pipeline can be reused in multiple pipelines. 
+
+The standard library can include standard definitions of these blocks that can be used in pipelines without requiring users to define them beforehand.
+
+As part of this RFC, we add the following blocks to the standard library:
+
+```
+block StandardTextFileInterpreter oftype TextFileInterpreter { }
+block StandardCSVInterpreter oftype CSVInterpreter { }
+block StandardXLSXInterpreter oftype XLSXInterpreter { }
+```
+
+## Drawbacks
+- This change introduces more structure for blocks but pipelines are still confusing, see Alternatives
```

Comment:
- With that enforced structure, users might loose flexibility to define things where they want (as an implication).

Leading to an alternative:
- leave as it is and document that the `pipe` syntax means instantiation and everything else is type definition
  - decided against by applying principle "Explicit modeling over hidden magic"; not mixing concepts makes things more explicit IMO

Reply:
Added this as drawback / alternative with the decision as described here.